### PR TITLE
revert(tests): Revert timeout that was added to jest acceptance

### PR DESCRIPTION
**Pull request by shruthilayaj**
Reverts test timeout set in https://github.com/getsentry/sentry/pull/30501

This was a bandaid fix to get some acceptance tests
passing that were causing jest timeouts. One of them
was fixed earlier, this PR addresses a few more that
were causing timeouts when this revert was made.
- Split the test into two here: alerts/incidentRules/details.spec.jsx
- Removed async modal stuff in favour of spying on
   openModal here: gridLayout/detail.spec.jsx

### CODE REVIEW

**Review by narsaynorath — APPROVED** [review]
Looks good to me, but maybe Kevan can comment more on the `incidentRules` test. Thanks for  updating the `gridLayout` test!

File: .github/workflows/acceptance.yml [hunk 1]
```diff
@@ -62,7 +62,7 @@ jobs:
           CI_NODE_TOTAL: 3
           CI_NODE_INDEX: ${{ matrix.instance }}
         run: |
-          JEST_TESTS=$(yarn -s jest --listTests --json) yarn test-ci --forceExit --testTimeout=10000
```
Comment (k-fish) [inline .github/workflows/acceptance.yml:65]: I'm glad we're rolling this back now, can you set a reminder for yourself for next week to check if the flake changes here?

**Review by k-fish — APPROVED** [review]
Cool, lets give it a shot. Worst case it goes back to being flaky and we roll it back next week.

File: .github/workflows/acceptance.yml (hunk 1 above)
Comment (shruthilayaj) [reply .github/workflows/acceptance.yml:65]: yup, will do 👍

**Comment by chadwhitacre** [comment]
Ftr I'm still getting bit by long-running tests here:

https://github.com/getsentry/sentry/runs/5190874023?check_suite_focus=true

**Pull request merged**